### PR TITLE
CLOUDP-118864: Split cloud manager and atlas decryptions

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

Split cloud manager and atlas decryptions 

_Jira ticket:_ CLOUDP-118864

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: build/ci/evergreen.yml [hunk 1]
```diff
@@ -1199,17 +1199,37 @@ tasks:
...
-          GCP_CREDENTIALS: ${logs_decrypt_gcp_credentials}
-          AWS_ACCESS_KEY: ${logs_decrypt_aws_access_key}
-          AWS_SECRET_ACCESS_KEY: ${logs_decrypt_aws_secret_access_key}
           LOCAL_KEY: ${logs_decrypt_local_key}
           KMIP_CA: ${logs_decrypt_kmip_ca}
           KMIP_CERT: ${logs_decrypt_kmip_cert}
+          MCLI_SERVICE: cloud-manager
+          E2E_TAGS: cloudmanager,decrypt
+  - name: atlas_decrypt_e2e
+    tags: [ "e2e","generic" ]
```
Comment (gssbzn) [inline build/ci/evergreen.yml:1208]: [q] does this test talk to AWS,GCP and AZURE? how generic is it? should it be moved to it's own target, it's it ok to run on every PR cost wise?

File: internal/decryption/encrypted_audit_log.go [hunk 2]
```diff
@@ -71,11 +71,17 @@ func (logLine *AuditLogLine) KeyProvider(opts *KeyProviderOpts) (keyproviders.Ke
 
 	switch *logLine.KeyStoreIdentifier.Provider {
 	case keyproviders.LocalKey:
+		if opts.Local == nil {
+			return nil, fmt.Errorf("keyProvider %s not supported", *logLine.KeyStoreIdentifier.Provider)
```
Comment (gssbzn) [inline internal/decryption/encrypted_audit_log.go:75]: [style] this and others like it, consider a sentinel error 
```go
var ErrInvalidKeyProvider = errors.New("unsupported key provider")
...
fmt.Errorf("%w: %s", ErrInvalidKeyProvider, *logLine.KeyStoreIdentifier.Provider)
```
this has the advantage later on we can do `errors.Is(err, ErrInvalidKeyProvider)` and catch it for all

File: internal/decryption/decryption.go [hunk 3]
```diff
@@ -33,10 +33,67 @@ func (s *DecryptSection) zeroLEK() {
 	}
 }
 
+type Decryption struct {
+	opts KeyProviderOpts
+}
+
+func NewDecryption(options ...func(d *Decryption)) *Decryption {
```
Comment (gssbzn) [inline internal/decryption/decryption.go:40]: [style] let's type this for easy of use
```go
type DecryptionOpt func(d *Decryption)
```
and use it around

**Review by gssbzn — DISMISSED** [review]
some small recommendations (non-blocking) and a question about running the test for log decryption as part of the PR suite

**Comment by fmenezes** [comment]
rfal @gssbzn

**Review by gssbzn — APPROVED** [review]
LGTM thanks for those changes

**Pull request merged**